sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -226,7 +226,10 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
             val naf = patchAggregateFunctionChildren(af) { x =>
               distinctAggChildAttrLookup.get(x).map(evalWithinGroup(id, _))
             }
-            (e, e.copy(aggregateFunction = naf, isDistinct = false))
+            val filterOpt = e.filter.map(_.transform {
+              case a: Attribute => distinctAggChildAttrLookup.getOrElse(a, a)
```

Comment:
what does it do?

Reply:
Because RewriteDistinctAggregates will create new attr to replace attr, so we need to update the attr in filter clause with the new one.